src/mlpack/methods/ann/layer/subview.hpp

```diff
@@ -56,10 +68,42 @@ class Subview
   template<typename InputType, typename OutputType>
   void Forward(InputType&& input, OutputType&& output)
   {
-    // Check if input has been selected as required.
-    if ((input.n_rows != (end - begin + 1)) && (end != 0))
+    size_t batchSize = input.n_cols / inSize;
+
+    // Check if subview parameters are within the indices of input sample.
+    endRow = ((endRow < input.n_rows) && (endRow >= beginRow))?
+        endRow : (input.n_rows - 1);
+    endCol = ((endCol < inSize) && (endCol >= beginCol)) ?
+        endCol : (inSize - 1);
+
+    output.set_size(
+        (endRow - beginRow + 1) * (endCol - beginCol + 1), batchSize);
+
+    size_t batchBegin = beginCol;
+    size_t batchEnd = endCol;
+
+    // Check whether the input is already in desired form.
+    if ((input.n_rows != ((endRow - beginRow + 1) *
+        (endCol - beginCol + 1))) || (input.n_cols != batchSize))
     {
-      output = arma::mat(&input(begin), end - begin + 1, input.n_cols, false);
+      for (size_t i = 0; i < batchSize; i++)
+      {
+        if (beginRow == 0 && endRow == (input.n_rows - 1))
+        {
+          // Matrix of contiguous elements.
+          output.col(i) = arma::mat(&input(batchBegin * input.n_rows),
+              (input.n_rows * (batchEnd - batchBegin + 1)), 1, false);
```

Comment:
So, I don't think there's a need to try and use any aliasing anymore, since `output.col(i) = ...` will always perform a copy.  I agree that we need to do it because the batch size is usually more than 1, but probably now we can simplify.

Reply:
Sorry for the late response. The notification just skipped me. Do you mean that we can just use submat() for both cases, in fact avoid having two cases? I didn't get what you meant by aliasing here.

Reply:
Right, that would simplify the code.

Reply:
Yeah, there's no need for the special constructor here, so we can always use `submat()`.